fix patch-next by adding missing script

Anas-github-acc opened the pull request:
The `patch-next.cjs` script is failing because two of its dependencies, `pack-util` and `build-native`, are missing their CommonJS versions, which are required for compatibility with the current Node.js environment. To fix this, i added the transpiled `pack-util.js` and `build-native.js` files.

Anas-github-acc committed:
fix:patch-next missing pack-util.js and build-native.js

ijjk commented:
Allow CI Workflow Run

- [x] approve CI run for commit: 98cd99561254cb7921299f657df688f48944ad78

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

bgw commented:
Hey @Anas-github-acc. We've been porting some of these scripts to typescript with ESM modules, under the assumption that future Next.js versions will be able to work well with that.

In the meantime, we've been using `tsx` to run similar scripts: https://github.com/vercel/next.js/blob/canary/scripts/pack-next.ts#L1 (see the comment at the top)

I don't want to accept a PR that adds back legacy CJS versions of this stuff, but I'd be willing to accept PRs fixing bugs in these scripts or porting them to TS and/or ESM.

FWIW, I only use the `pack-next` script instead without the `--tar` argument for my work in Turbopack.

bgw reviewed (changes requested):
See previous comment

Anas-github-acc commented:
@bgw Thankyou for reviewing my changes, and i agree that adding back legacy CJS is a dumb idea at this point

I am thinking of porting `patch-next` to typescript with ESM.

Its worth adding 'patch-next script', as it's quite faster when we want to quickly test a change without running `pnpm install` and `pack-next`

Anas-github-acc committed:
ported patch-next.cjs to typescript

Anas-github-acc committed:
some fixes

Anas-github-acc commented:
Hi @bgw

I've ported the `patch-next` to TypeScript. Could you please review my updates and let me know if there’s anything that needs adjustment or improvement? 

Thanks

Anas-github-acc committed:
fixed build and native-build flags

Anas-github-acc committed:
Update patch-next

Anas-github-acc committed:
Update scripts/patch-next.ts

Co-authored-by: graphite-app[bot] <96075541+graphite-app[bot]@users.noreply.github.com>

Anas-github-acc committed:
Update patch-next

Anas-github-acc committed:
Merge branch 'fix/patch-next_fail' of https://github.com/Anas-github-acc/next.js into fix/patch-next_fail

bgw committed:
A few fixes and opinionated tweaks to patch-next

- Rename `--native-build` to `--build-native` to match our naming
  elsewhere, but alias `--native-build` (sure, why not?)

- Use GNU-style counted `-v` arguments for verbosity, remove the short
  version alias, which conflicts and isn't very useful for a script
  like this.

- Fix how `buildNative` is called. We no longer do the horrible
  import-time side-effect stuff, and you now must call the default
  export. Fix positional argument passthrough as well.

- Move `example` into the default yargs command builder so that it works
  correctly, add an extra example about passing in the `--release` flag

- Apply `hideBin` to pack-next as well.

bgw commented:
This looks awesome @Anas-github-acc. It's clear a lot of love and effort went into this.

I tried it out and made a few small tweaks and fixes in c6a682688aa8b2cb6ace8f7cb4d21693b37e1291 (I hope that's okay), and I'll merge once CI finishes.

bgw commented on scripts/patch-next.ts:
```diff
@@ -0,0 +1,178 @@
+import { hideBin } from 'yargs/helpers'
+import path from 'path'
+
+import { NEXT_DIR, exec, execFn, packageFiles } from './pack-util.js'
+import buildNative from './build-native.js'
```
This file is intended to be run with `tsx`.

Anas-github-acc commented:
Thankyou so much @bgw  i'm happy you liked it. I really enjoyed working on this.
I appreciate you added the final tweaks before it got merged
